doc/cephfs/purge-queue.rst

```diff
@@ -0,0 +1,83 @@
+============
+Purge Queue
+============
+
+MDS maintains a data structure known as **Purge Queue** which is responsible
+for managing and executing the sequential deletion of files.
+There is a Purge queue for every MDS rank. Purge queues consist of purge items
+which contain nominal information from the inodes such as size and the layout
+(i.e. all other un-needed metadata information is discarded making it
+independent of all metadata structures).
+
+Deletion process
+================
+
+When a client requests deletion of a directory (say ``rm -rf``):
+
+- MDS queues the files and subdirectories (purge items) from journal in the
```

Comment:
So, there is a purge queue journal (`pq`). Its a bit unclear which journal is being referred to here.

Reply:
The journal here comes from `osdc/Journaler.h"`. Since this is MDS's side, would it be correct to call it MDS Journal?

Reply:
So, there is MDS metadata journal (mdlog) and the MDS purge queue journal (pq). Both of course use osdc/Journaler.h class, but my question here is which of the above two journals is being referred.

Reply:
It's purge queue journal https://github.com/ceph/ceph/blob/5061b31102daaa52b56749252e81aaccb8a3ab83/src/mds/PurgeQueue.cc#L129-L131

Reply:
Right. So, let's mention that explicitly in the doc.

Reply:
@vshankar, I'll get this in the document when I get back to my office. It'll be in today.

Reply:
At ease @zdover23. I have requested one small update from @dparmar18 and then its good to go 👍

Reply:
Roger wilco, @vshankar.

Reply:
@vshankar [PTAL](https://github.com/ceph/ceph/compare/0230a8b11ebf90b1bfa4fb797afe56a7eadc11a9..ae9277398840bf8576ea5a8c4a2ba4e23f8b9613)